modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbTableStorage.java

```diff
@@ -0,0 +1,316 @@
+
+    /** Rocks DB instance. */
+    private volatile TransactionDB db;
+
+    /** RocksDb database options. */
+    private volatile DBOptions dbOptions;
+
+    /** Transactional RocksDb options. */
+    private volatile TransactionDBOptions txDbOptions;
+
+    /** Write options. */
+    private final WriteOptions writeOptions = new WriteOptions().setDisableWAL(true);
+
+    /** Read options for regular reads. */
+    private final ReadOptions readOptions = new ReadOptions();
+
+    /** Read options for reading persisted data. */
+    private final ReadOptions persistedTierReadOptions = new ReadOptions().setReadTier(PERSISTED_TIER);
+
+    /** Database path. */
+    private final Path dbPath;
+
+    /** Partition storages. */
+    private volatile AtomicReferenceArray<TxStateStorage> storages;
+
+    /** Table configuration. */
+    private final TableConfiguration tableCfg;
+
+    /** RocksDB flusher instance. */
+    private volatile RocksDbFlusher flusher;
+
+    /** Prevents double stopping the storage. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
+    /** Busy lock to stop synchronously. */
+    final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    private final ScheduledExecutorService scheduledExecutor;
+
+    /** Thread pool to execute after-flush actions. */
+    private final ExecutorService threadPool;
+
+    /** Supplier for the value of delay for scheduled database flush. */
+    private final IntSupplier flushDelaySupplier;
+
+    /**
+     * Constructor.
+     *
+     * @param tableCfg Table configuration.
+     * @param dbPath Database path.
+     * @param scheduledExecutor Scheduled executor.
+     */
+    public TxStateRocksDbTableStorage(
+            TableConfiguration tableCfg,
+            Path dbPath,
+            ScheduledExecutorService scheduledExecutor,
+            ExecutorService threadPool,
+            IntSupplier flushDelaySupplier
+    ) {
+        this.tableCfg = tableCfg;
+        this.dbPath = dbPath;
+        this.scheduledExecutor = scheduledExecutor;
+        this.threadPool = threadPool;
+        this.flushDelaySupplier = flushDelaySupplier;
+    }
+
+    /**
+     * Checks that a passed partition id is within the proper bounds.
+     *
+     * @param partitionId Partition id.
+     */
+    private void checkPartitionId(int partitionId) {
+        if (partitionId < 0 || partitionId >= storages.length()) {
+            throw new IllegalArgumentException(S.toString(
+                "Unable to access partition with id outside of configured range",
+                "table", tableCfg.name().value(), false,
+                "partitionId", partitionId, false,
+                "partitions", storages.length(), false
+            ));
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public TxStateStorage getOrCreateTxStateStorage(int partitionId) throws StorageException {
```

Comment:
Title says it's "getOrCreate", but implementation does unconditional "create". Something's wrong here